### ISSUE TITLE
Chart versions management

### ISSUE THREAD

**Issue by JorTurFer** [kedacore/charts#363]
Until now, we usually release a KEDA version with the chart without hotfix releases, this helped us because all changes in (charts) main branch are released with KEDA, but since v2.9 we have decided to do hotfix releases at any time. 
This can be a pain with helm because we don't have any release branch, all the releases are from main and all the charts are served from main, making really complex a hotfix release without including other changes. A hacky solution could be generating the chart from the latest tag and committing it into main but this could be problematic.

We should think about this scenario and look for a consistent solution for the future. Maybe we can move the helm charts to any other place (if there is any other free helm storage), or maybe we can move the code to other repo/branch, keeping current main branch as release branch only for having the artefacts and not the code. With this approach, we could have the development branch, doing the releases from it, and as part of the release process, adding the chart to the branch used for serving the charts

**Comment by zroubalik** [comment]
I think we should follow the same procedure that's present on core, to create a separate branch for each minor releases, ie: `release/v2.9`, release/v2.10` etc.

Then create the release, tag and package against this branch and then copy over the package to main for the actual release.

**Comment by JorTurFer** [comment]
Don't you think that having a branch only for the artifacts is better? I think we can automatize the push to main on pushed tags.
The process would be:
We work on branch XXX, on every minor release we create the branch component-version (e.g: keda-v2.9) and a tag inside it. On every tag pushed, we can execute `helm package`, upgrade the index and open a PR against the branch (or maybe pushing it directly) with the artifacts.
We can continue working on branch XXX, and cherry-picking hotfixes to the release branch.

WDYT?

**Comment by tomkerkhove** [comment]
I'm not a fan of automation doing all of this though, I still prefer if an actual human does this to avoid shipping packages we don't want to.

The only exception is using Git tags (which map to GitHub Releases).

> I think we should follow the same procedure that's present on core, to create a separate branch for each minor releases, ie: `release/v2.9`, release/v2.10` etc.
> 
> Then create the release, tag and package against this branch and then copy over the package to main for the actual release.

I agree on this; main is where everything should be at imo.

**Comment by JorTurFer** [comment]
Don't you like the idea of having a branch only for serving the helm packages? I mean, instead of serving them from the main branch, having a `helm` branch where we push only the `docs` folder. We could split binaries from the code and treat that branch as a pseudo external thing from the code. E.g: we use GHCR for images, but they aren't in the code, they are near to it, but not in the same place

**Comment by tomkerkhove** [comment]
Personally no, I think it's best to keep it simple and use main as the default as hotfixes should be minimized. If we minimize them, another branch just adds more churn as we'll need two PRs or things are spread in various places.

I personally don't see much need to make it more complex for now.

**Comment by nunoadrego** [comment]
> E.g: we use GHCR for images, but they aren't in the code, they are near to it, but not in the same place

I saw [this post](https://trstringer.com/helm-charts-github-container-registry/) and remembered this discussion.

**Comment by JorTurFer** [comment]
> I saw [this post](https://trstringer.com/helm-charts-github-container-registry/) and remembered this discussion.

Yeah, using `oci` you can use multiple registries, but AFAIK, you lose the options to browse the repository, doing search, etc...

**Comment by JorTurFer** [comment]
Actually, I think that maybe we should start building OCI charts and directly deprecate the old method. using OCI our minor releases will significantly easier I'd say, as we will ship the chart from the release branch and that's all

**Comment by DrFaust92** [comment]
@JorTurFer , Any movement on this?
